packages/api/models/granules.js

```diff
@@ -576,60 +579,43 @@ class Granule extends Manager {
   }
 
   /**
-   * Parse a Cumulus message and build granule records for the embedded granules.
+   * Store a granule record.
    *
-   * @param {Object} cumulusMessage - A Cumulus message
-   * @returns {Promise<Array<Object>>} - An array of granule records
+   * @param {Object} granuleRecord - A granule record.
+   * @returns {Promise<Object|undefined>}
    */
-  static async _getGranuleRecordsFromCumulusMessage(cumulusMessage) {
-    const granules = getMessageGranules(cumulusMessage);
-    if (!granules) {
-      log.info(`No granules to process in the payload: ${JSON.stringify(cumulusMessage.payload)}`);
-      return [];
-    }
+  async _storeGranuleRecord(granuleRecord) {
+    const mutableFieldNames = this._getMutableFieldNames(granuleRecord);
+    const updateParams = this._buildDocClientUpdateParams({
+      item: granuleRecord,
+      itemKey: { granuleId: granuleRecord.granuleId },
+      mutableFieldNames,
+    });
 
-    const executionArn = getMessageExecutionArn(cumulusMessage);
-    const executionUrl = StepFunctionUtils.getExecutionUrl(executionArn);
+    // Only allow "running" granule to replace completed/failed
+    // granule if the execution has changed
+    if (granuleRecord.status === 'running') {
+      updateParams.ConditionExpression = '#execution <> :execution';
+    }
 
-    let executionDescription;
     try {
-      executionDescription = await StepFunctions.describeExecution({ executionArn });
+      return await this.dynamodbDocClient.update(updateParams).promise();
     } catch (error) {
-      log.error(`Could not describe execution ${executionArn}`, error);
+      if (isConditionalCheckException(error)) {
```

Comment:
I thought this made sense when I wrote it, but looking closer, the previous code in `_validateAndStoreGranuleRecord` treated all errors the same and caught them, where they were a conditional check exception or not. So these conditional check exceptions should probably be thrown, since this method is used in the RDS transaction to write to Dynamo and we want to fail both writes if either write fails.

Reply:
Addressed here: bce1e9370a57e4bff67de02126e348a4e7adc6b9